pkg/eestream/ranger/reader.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package ranger
+
+import (
+	"bytes"
+	"io"
+)
+
+// A Ranger is a flexible data stream type that allows for more effective
+// pipelining during seeking. A Ranger can return multiple parallel Readers for
+// any subranges.
+type Ranger interface {
+	Size() int64
+	Range(offset, length int64) io.Reader
+}
+
+// FatalReader returns a Reader that always fails with err.
+func FatalReader(err error) io.Reader {
+	return &fatalReader{Err: err}
+}
+
+type fatalReader struct {
+	Err error
+}
+
+func (f *fatalReader) Read(p []byte) (n int, err error) {
+	return 0, f.Err
+}
+
+// ByteRanger turns a byte slice into a Ranger
+type ByteRanger []byte
+
+func (b ByteRanger) Size() int64 { return int64(len(b)) }
+
+func (b ByteRanger) Range(offset, length int64) io.Reader {
```

Comment:
Same question here about return value in case of error. `nil` is an obvious choice, but the actual error is lost. So maybe `(io.Reader, error)`?

Reply:
here, yeah, there should probably be an error, buuut, io.Reader can return errors, so I made the FatalError() wrapper. if this method has a non-nil err, return FatalReader(err).